### ISSUE TITLE
rustc wrongly reports conflicting implementation

### ISSUE THREAD

**Issue by ngolin** [rust-lang/rust#62582]
**`String` is not a `Copy`, there should be no conflicts.**
```rust
struct S<V>(V);

trait T<V> {
  fn m(&self) -> V;
}

impl<V: Copy> T<V> for S<V> {
  fn m(&self) -> V {
    self.0
  }
}

impl T<String> for S<String> {
  fn m(&self) -> String {
    String::from("String")
  }
}
```

```
error[E0119]: conflicting implementations of trait `T<std::string::String>` for type `S<std::string::String>`:
  --> src/main.rs:13:1
   |
7  | impl<V: Copy> T<V> for S<V> {
   | --------------------------- first implementation here
...
13 | impl T<String> for S<String> {
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ conflicting implementation for `S<std::string::String>`
   |
   = note: upstream crates may add new impl of trait `std::marker::Copy` for type `std::string::String` in future versions
```

**Comment by ngolin** [comment]
I missed the note at the bottom.
